backend/src/main/java/ai/verta/modeldb/configuration/ReconcilerInitializer.java

```diff
@@ -23,10 +33,13 @@ public class ReconcilerInitializer {
   public static UpdateProjectTimestampReconcile updateProjectTimestampReconcile;
   public static SendEventsWithCleanUp sendEventsWithCleanUp;
 
-  public static void initialize(
-      Config config, ServiceSet services, DAOSet daos, FutureJdbi futureJdbi, Executor executor) {
+  @Bean
+  @ConditionalOnBean(DAOSet.class) // the DAOSet is explicitly not valid when doing migrations
```

Comment:
this is the important bit to make sure we don't try to create this bean while doing migrations. It's a bit of a hack. I recommend that we create a bean that is specifically crafted to be the thing that we use for this, rather than hacking it via the `null` return in the creation of the `DAOSet` bean at the moment.